### PR TITLE
Blogger Plan: Domains: Show upgrade page when adding more then 1 domain

### PR DESCRIPTION
**Pull request by matthusby**
The Blogger plan only comes with 1 `.blog` domain, if the user is trying to add more then 1 we should take them to the plan upgrades.

#### Testing instructions

* D23046-code handles the backend part of this
* On a blogger plan that has a domain when you try to add another one you will be prompted to upgrade

### CODE REVIEW

File: client/my-sites/domains/domain-search/domain-search.jsx [hunk 1]
```diff
@@ -122,6 +126,15 @@ class DomainSearch extends Component {
...
 		} );
 		const { domainRegistrationMaintenanceEndTime } = this.state;
+
+		const { domains } = this.props;
+		const products = get( this.props, 'cart.products', [] );
+		const plan = get( this.props, 'selectedSite.plan', false );
+
+		const domainInCart = products.some( isDomainMapping ) || products.some( isDomainRegistration );
+		const isBloggerPlan = isBlogger( plan ) || products.some( isBlogger );
+		const atDomainLimit = isBloggerPlan && ( domains.length > 0 || domainInCart );
```
[the remark this answers is not in the document]
Comment (matthusby) [reply client/my-sites/domains/domain-search/domain-search.jsx]: Oh good catch, I wasn't filtering out wpcom domains from the list that gets checked.  I just updated the diff

File: client/my-sites/domains/domain-search/domain-search.jsx [hunk 2]
```diff
@@ -122,6 +126,15 @@ class DomainSearch extends Component {
 			'domain-search-page-wrapper': this.state.domainRegistrationAvailable,
 		} );
 		const { domainRegistrationMaintenanceEndTime } = this.state;
+
+		const domains = this.props.domains.filter( domain => domain.type !== 'WPCOM' );
+		const products = get( this.props, 'cart.products', [] );
+		const plan = get( this.props, 'selectedSite.plan', false );
+
+		const domainInCart = products.some( isDomainMapping ) || products.some( isDomainRegistration );
```
[the remark this answers is not in the document]
Comment (matthusby) [reply client/my-sites/domains/domain-search/domain-search.jsx]: D23046-code Has been fixed and deployed, anything else stopping this from going out?

**Comment by niranjan-uma-shankar** [comment]
Thanks for working on the backend changes @matthusby. 

LGTM, once the tests pass. Looks like the failures are unrelated to this PR though. Maybe a rebase and restarting the tests would help.

File: client/my-sites/domains/domain-search/domain-search.jsx [hunk 3]
```diff
@@ -122,6 +126,15 @@ class DomainSearch extends Component {
 			'domain-search-page-wrapper': this.state.domainRegistrationAvailable,
 		} );
 		const { domainRegistrationMaintenanceEndTime } = this.state;
+
+		const domains = this.props.domains.filter( domain => domain.type !== 'WPCOM' );
```
Comment (southp) [inline client/my-sites/domains/domain-search/domain-search.jsx]: @matthusby FYI there is a handy constant for exactly this scenario in [lib/domains/constants](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/constants.js#L7) An example usage can be found here: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/domains/domain-management/edit/wpcom-domain.jsx#L32.

Comment (matthusby) [reply client/my-sites/domains/domain-search/domain-search.jsx]: Thanks! I just opened https://github.com/Automattic/wp-calypso/pull/30703 to address this :+1:

**Pull request merged**